research_fish_analysis.py

```diff
@@ -1,288 +1,288 @@
+ 
+ 
+def import_xls_to_df(filename, name_of_sheet): 
+    # Set up logging 
+    logger = logging.getLogger(__name__) 
+    logger.info('Importing data...') 
+ 
+    return pd.read_excel(filename,sheetname=name_of_sheet) 
+ 
+ 
+def add_column5(dataframe,newcol): 
+    """ 
+    Adds a new column of NaNs called newcol 
+    :params: a dataframe and column name 
+    :return: a dataframe with a new column 
+    """ 
+     
+    # Set up logging 
+ 
+    logger = logging.getLogger(__name__) 
+ 
+    logger.info('Adding a column...') 
+ 
+ 
+ 
+    dataframe[newcol] = np.nan 
+ 
+    return dataframe 
+ 
+ 
+#def add_column2(dataframe,newcol): 
+#    """ 
+#    Adds a new column of NaNs called newcol 
+#    :params: a dataframe and column name 
+#    :return: a dataframe with a new column 
+#    """ 
+#    # Set up logging 
+#    logger = logging.getLogger(__name__) 
+#    logger.info('Adding a column...') 
+#    dataframe[newcol] = np.nan 
+#    return dataframe 
+ 
+ 
+def clean_data2(dataframe,colname): 
+    logger = logging.getLogger(__name__) 
+    logger.info('Cleaning data...') 
+     
+    l = len(dataframe) 
+ 
+    dataframe = dataframe[(dataframe['Type of Tech Product'] == 'Software') | (dataframe['Type of Tech Product'] == 'Grid Application') | (dataframe['Type of Tech Product'] == 'e-Business Platform') | (dataframe['Type of Tech Product'] == 'Webtool/Application')] 
+ 
+    t = len(dataframe) 
+     
+    dataframe.drop_duplicates(subset = ['Tech Product'], keep = 'first', inplace = True) 
+ 
+    length_dupes = len(dataframe) 
+     
+    lost_years = ['2006', '2007', '2008', '2009', '2010', '2011'] 
+    for year in lost_years: 
+        dataframe.drop(dataframe[dataframe[colname] == year].index, inplace = True) 
+ 
+    y = len(dataframe) 
+     
+    for i, row in dataframe.iterrows(): 
+        try: 
+            int(dataframe[colname][i]) 
+        except:  
+            dataframe[colname][i] = np.nan 
+     
+    dataframe.dropna(subset=[colname], inplace=True) 
+     
+    length_final = len(dataframe) 
+ 
+    logger.info("Records dropped during tech product cleaning: " + repr(l - t)) 
+    logger.info("Records dropped during duplicate cleaning: " + repr(t - length_dupes)) 
+    logger.info("Records dropped when cleaning years outside 2012-2016: " + repr(length_dupes - y))        
+    logger.info("Records dropped during non-valid-year cleaning: " + repr(y - length_final)) 
+    logger.info("Records left in cleaned data set: " + repr(length_final)) 
+     
+    return dataframe 
+     
+ 
+def produce_count(dataframe, colname): 
+    # Set up logging 
+    logger = logging.getLogger(__name__) 
+    logger.info('Producing a count...')     
+     
+    dataframe = pd.DataFrame(dataframe[colname].value_counts()) 
+     
+    # Add a column for percentages 
+    dataframe['percentage'] = dataframe[colname]/dataframe[colname].sum() 
+     
+    return dataframe 
+ 
+     
+def produce_count_and_na(dataframe, colname): 
```

Comment:
It wasn't clear to me from the name exactly what this function does (I had to look at the actual code). How about renaming it to `produce_count_including_na_values`?